### PR TITLE
Revision one of external packages docs

### PR DESCRIPTION
**Pull request by benrudolph**
@dannyroberts 

cc: @dimagi/js-team

### CODE REVIEW

File: external-packages.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+### Bower packages
+
+Bower packages can be installed from a variety of sources, including a registered bower package (a repo that has a `bower.json` file defined`), a Github shorthand (<user or org>/<repo_name>), a Github URL, or just a plain URL that points to a javascript file.
+
+When you install a package, it will be installed in a directory called `bower_components`. For example if you were to run `bower install jquery`, you would find a directory `bower_components/jquery`.
+
+### Specifying packages in `bower.json`
+
+To ensure a package gets installed for a project, you must specify it in the `bower.json` file. This is equivalent to the `requirements.txt` file for `pip`. When specifying a bower package you can use many techniques. Here are a few examples:
```
Comment (orangejenny) [inline external-packages.md]: I'd add that `bower update` is the equivalent of `pip install...`

**Comment by esoergel** [comment]
(branch has conflicts)

File: external-packages.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+"jquery-other": "jquery#1.2.0"
+
+// This will install jquery from a github hash
+"jquery-github": "jquery/jquery#44cb97e0cfc8d3e62bef7c621bfeba6fe4f65d7c"
+
+```
+
+To generalize, an install declaration looks like this:
+```
+<name>=<package>#<version>
```
Comment (esoergel) [inline external-packages.md]: Should be a `:` instead of a `=`

File: external-packages.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+| URL | http://example.com/script.js |
+
+There are more, but those are the important ones. Find the others [here](http://bower.io/docs/api/#install)
+
+A version can be any of these things:
+
+| Type | Example |
+|---|---|
+| semver | `#1.2.3` |
+| version range | `#~1.2.3` |
```
Comment (esoergel) [inline external-packages.md]: What does this range mean?

Comment (benrudolph) [inline external-packages.md]: it's all part of the semantic versioning ranging. the `~` allows it to vary on the last digit (bug fixes). there are others as well: https://nodesource.com/blog/semver-tilde-and-caret/

**Comment by benrudolph** [comment]
thanks for feedback guys. updated!

File: external-packages.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# External Packages
+
+## Bower
+
+Bower can manage components that contain HTML, CSS, JavaScript, fonts or even image files. Bower doesn’t concatenate or minify code or do anything else - it just installs the right versions of the packages you need and their dependencies.
+
+### Bower packages
+
+Bower packages can be installed from a variety of sources, including a registered bower package (a repo that has a `bower.json` file defined`), a Github shorthand (<user or org>/<repo_name>), a Github URL, or just a plain URL that points to a javascript file.
```
Comment (millerdev) [inline external-packages.md]: extra `` ` `` after "file defined"

Comment (millerdev) [inline external-packages.md]: `(<user or org>/<repo_name>)` does not display correctly in markdown version. it comes out as "(/)"

File: external-packages.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+A version can be any of these things:
+
+| Type | Example |
+|---|---|
+| semver | `#1.2.3` |
+| version range | `#~1.2.3` |
+| Git tag | `#<git tag>` |
+| Git commit | `#<commit sha>` |
+| Git branch | `#<branch>` |
```
Comment (millerdev) [inline external-packages.md]: I think the leading `#` is also optional

File: external-packages.md (hunk 4 above)
Comment (benrudolph) [inline external-packages.md]: thanks will update!

File: external-packages.md (hunk 5 above)
Comment (benrudolph) [inline external-packages.md]: decided to leave it since that's the way it's presented in the docs

**Pull request merged**